### ISSUE TITLE
IoU Loss should only reduce over 1 axis by default.

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#762]
Keras losses reduce over one axis by default - currently IoU is reducing over two.  This will make it impossible for us to mask out background boxes.  To get around this, we should only reduce over one axis instead of two in IoU loss.

This means that after the IoU is computed you should have [batch_size, num_boxes] returned from the loss

**Comment by LukeWood** [comment]
@quantumalaviya

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
